Signal/src/environment/OWSPreferences.m

```diff
@@ -0,0 +1,317 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWSPreferences.h"
+#import "TSStorageHeaders.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+double const OWSPreferencesDefaultCallStreamDESBufferLevel = 0.5;
+NSString *const OWSPreferencesSignalDatabaseCollection = @"SignalPreferences";
+
+NSString *const OWSPreferencesKeyCallStreamDESBufferLevel = @"CallStreamDesiredBufferLevel";
+NSString *const OWSPreferencesKeyScreenSecurity = @"Screen Security Key";
+NSString *const OWSPreferencesKeyEnableDebugLog = @"Debugging Log Enabled Key";
+NSString *const OWSPreferencesKeyNotificationPreviewType = @"Notification Preview Type Key";
+NSString *const OWSPreferencesKeyHasSentAMessage = @"User has sent a message";
+NSString *const OWSPreferencesKeyHasArchivedAMessage = @"User archived a message";
+NSString *const OWSPreferencesKeyPlaySoundInForeground = @"NotificationSoundInForeground";
+NSString *const OWSPreferencesKeyHasRegisteredVoipPush = @"VOIPPushEnabled";
+NSString *const OWSPreferencesKeyLastRecordedPushToken = @"LastRecordedPushToken";
+NSString *const OWSPreferencesKeyLastRecordedVoipToken = @"LastRecordedVoipToken";
+NSString *const OWSPreferencesKeyCallKitEnabled = @"CallKitEnabled";
+NSString *const OWSPreferencesKeyCallKitPrivacyEnabled = @"CallKitPrivacyEnabled";
+NSString *const OWSPreferencesKeyCallsHideIPAddress = @"CallsHideIPAddress";
+NSString *const OWSPreferencesKeyHasDeclinedNoContactsView = @"hasDeclinedNoContactsView";
+NSString *const OWSPreferencesKeyIOSUpgradeNagVersion = @"iOSUpgradeNagVersion";
+NSString *const OWSPreferencesKeyAreReadReceiptsEnabled = @"areReadReceiptsEnabled";
+
+@implementation OWSPreferences
+
+- (instancetype)init
+{
+    self = [super init];
+    if (!self) {
+        return self;
+    }
+
+    OWSSingletonAssert();
+
+    return self;
+}
+
+#pragma mark - Helpers
+
+- (void)clear {
+    @synchronized(self) {
+        NSString *appDomain = NSBundle.mainBundle.bundleIdentifier;
+        [NSUserDefaults.standardUserDefaults removePersistentDomainForName:appDomain];
+    }
+}
+
+- (nullable id)tryGetValueForKey:(NSString *)key
+{
+    ows_require(key != nil);
+    return [TSStorageManager.sharedManager objectForKey:key inCollection:OWSPreferencesSignalDatabaseCollection];
+}
+
+- (void)setValueForKey:(NSString *)key toValue:(nullable id)value
+{
+    ows_require(key != nil);
+
+    [TSStorageManager.sharedManager setObject:value forKey:key inCollection:OWSPreferencesSignalDatabaseCollection];
+}
+
+#pragma mark - Specific Preferences
+
+- (NSTimeInterval)getCachedOrDefaultDesiredBufferDepth
+{
```

Comment:
Pretty sure this is redphone and can go away.

Reply:
Removed.